### PR TITLE
Add shared files screen

### PR DESCRIPTION
**Pull request by adasiewiczr**
Screenshot 2020-12-17 at 16 11 04

Screenshot 2020-12-17 at 16 29 07

### Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Changelog updated with client-facing changes
- [ ] New code is covered by unit tests
- [x] Comparison screenshots added for visual changes
- [x] Reviewers added

### CODE REVIEW

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt [hunk 1]
```diff
@@ -447,19 +447,40 @@ public class ChatClient internal constructor(
         return api.searchMessages(request)
     }
 
-    public fun getFileAttachments(channelType: String, channelId: String, offset: Int, limit: Int): Call<List<Attachment>> =
+    public fun getFileAttachments(
```
Comment (adasiewiczr) [inline stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt:450]: Those methods weren't released yet so this is not a breaking change

**Review by ogkuzmin — DISMISSED** [review]

File: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/info/shared/files/ChatInfoSharedFilesViewHolders.kt [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    attachmentClickListener: ChatInfoSharedFilesAdapter.AttachmentClickListener?
+) : BaseViewHolder<SharedAttachment.AttachmentItem>(binding.root) {
+
+    private lateinit var attachmentItem: SharedAttachment.AttachmentItem
+
+    init {
+        binding.fileItemContainer.setOnClickListener { attachmentClickListener?.onClick(attachmentItem) }
+    }
+
+    @InternalStreamChatApi
```
Comment (zsmb13) [inline stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/info/shared/files/ChatInfoSharedFilesViewHolders.kt:37]: This should not happen in the sample outside of the component browser, anything that's needed to build the sample should be public in the SDK or implemented within the sample if necessary

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt [hunk 3]
```diff
@@ -987,7 +1009,10 @@ public class ChatClient internal constructor(
         public val isInitialized: Boolean
             get() = instance != null
 
-        public fun isValidRemoteMessage(remoteMessage: RemoteMessage, defaultNotificationConfig: NotificationConfig = NotificationConfig()): Boolean =
+        public fun isValidRemoteMessage(
+            remoteMessage: RemoteMessage,
+            defaultNotificationConfig: NotificationConfig = NotificationConfig()
+        ): Boolean =
```
Comment (zsmb13) [inline stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt:1015]: This method should just have a proper body at this point, doesn't make sense for it to have an expression body

File: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/info/shared/files/ChatInfoSharedFilesViewHolders.kt (hunk 2 above)
Comment (zsmb13) [inline stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/info/shared/files/ChatInfoSharedFilesViewHolders.kt:37]: This should not happen in the sample outside of the component browser, anything that's needed to build the sample should be public in the SDK or implemented within the sample if necessary

**Review by zsmb13 — CHANGES_REQUESTED** [review]

File: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/info/shared/files/ChatInfoSharedFilesViewHolders.kt (hunk 2 above)
Comment (adasiewiczr) [reply stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/info/shared/files/ChatInfoSharedFilesViewHolders.kt:37]: True, I didn't want to copy all file types related resources to sample app. I'll make `UiUtils::getIcon` and `MediaStringUtil::convertFileSizeByteCount ` public for now

**Review by zsmb13 — APPROVED** [review]
LGTM

**Pull request merged**